### DISCUSSION TITLE
Efficient way of determining size and type of image using Magick++

### DISCUSSION THREAD

**Discussion by adhawkins** [ImageMagick/ImageMagick#5954] [Help] [answered]
### ImageMagick version

6.9.11.60+dfsg-1.3

### Operating system, version and so on

Debian 11.6

### Description

Is there a 'quick' way using Magick++ to get the dimensions and type (i.e. jpeg, png etc.) of an image from a block of data? I'm doing the following at the moment, and it seems relatively 'slow' (presumably because it's having to fully load and decompress the image). Was wondering if there might be a 'quicker' way?

```
			Magick::Blob Blob(m_Data, m_Length);
			Magick::Image Image(Blob);
			auto Magick = Image.magick();
			if (!Magick.empty())
			{
				try
				{
					Magick::CoderInfo CoderInfo(Magick);

					auto Size = Image.size();

					m_Width = Size.width();
					m_Height = Size.height();

					if (m_MimeType.empty())
						m_MimeType = CoderInfo.mimeType();
				}
				catch (const std::exception &e)
				{
					std::stringstream os;
					os << "ImageMagick exception: '" << e.what() << "' - '" << Magick << "'";
					CErrorLog::Log(os.str());
				}
			}
			else
			{
				std::stringstream os;
				auto Size = Image.size();
				os << "ImageMagick 'Magick' is empty - " << Size.width() << " x " << Size.height();
				CErrorLog::Log(os.str());
			}
```

Any assistance appreciated.

Andy

**Comment by dlemstra** [comment, marked as answer]
You could do this instead:

```c++
Magick::Image Image();
Image.ping(Blob);
```

**Reply by adhawkins** [reply]
Oooh, that sounds like a good plan. Trying now.

Andy

**Reply by adhawkins** [reply]
Much better, gone from 58% of runtime (it gets called for 14 different images) to 1%!

Thanks a lot for the advice, much appreciated!

Andy
